src/Compilers/VisualBasic/Portable/Symbols/ObsoleteAttributeHelpers.vb

```diff
@@ -40,52 +45,81 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
         ''' hierarchy is Obsolete.
         ''' </returns>
         Friend Shared Function GetObsoleteContextState(symbol As Symbol, Optional forceComplete As Boolean = False) As ThreeState
-            If symbol Is Nothing Then
-                Return ThreeState.False
-            End If
-
-            If forceComplete Then
-                symbol.ForceCompleteObsoleteAttribute()
-            End If
+            While symbol IsNot Nothing
+                ' For property or event accessors, check the associated property or event instead.
+                If symbol.IsAccessor() Then
+                    symbol = DirectCast(symbol, MethodSymbol).AssociatedSymbol
```

Comment:
Updated to check accessor before associated symbol for both properties and events. (Note that C# reports `ERR_AttributeNotOnAccessor` when an obsolete attribute is applied to an accessor while VB reports `ERR_ObsoleteInvalidOnEventMember` when an obsolete attribute is applied to an event accessor only.)